plugins/web/opentelemetry-plugin-user-interaction/src/userInteraction.ts

```diff
@@ -165,6 +170,61 @@ export class UserInteractionPlugin extends BasePlugin<unknown> {
     }
   }
 
+  /**
+   * Returns true iff we should use the patched callback; false if it's already been patched
+   */
+  private addPatchedListener(
+    on: HTMLElement,
+    type: string,
+    listener: Function,
+    wrappedListener: Function
+  ): boolean {
+    let listener2Type = this._wrappedListeners.get(listener);
+    if (!listener2Type) {
+      listener2Type = new Map();
```

Comment:
Is this going to be coerced into a `WeakMap` when you set it on `_wrappedListeners`?

Reply:
No, though making the HTMLElement reference weak is a good idea.  Unfortunately I don't see a way to do it since there are no iterators/size/isEmpty methods available on WeakMap and without it we'd *definitely* leak.  I'll ponder this but I'd like to get this fix in as it is since it's clearly better than the current breaking of functional behavior.

Reply:
Awesome, thanks for the clarification!